File: java_projects/cleaned_final_projects/commons-fileupload/src/test/java/org/apache/commons/fileupload/portlet/MockPortletActionRequest.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.apache.commons.fileupload.portlet;

import org.apache.commons.fileupload.FileUploadBase;

import java.io.BufferedReader;
import java.io.ByteArrayInputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.UnsupportedEncodingException;
import java.util.Arrays;
import java.util.Collections;
import java.util.Enumeration;
import java.util.HashMap;
import java.util.Hashtable;
import java.util.Locale;
import java.util.Map;

/**
 * Mock class for tests. Implements an {@link ActionRequest}.
 *
 * @see PortletFileUploadTest
 * @since 1.4
 */
@SuppressWarnings("rawtypes") // because of the portlet ActionRequest API does not use generics
public class MockPortletActionRequest {

    private final Hashtable<String, Object> attributes = new Hashtable<String, Object>();

    private final Map<String, String> parameters = new HashMap<String, String>();

    private String characterEncoding;
    private final int length;
    private final String contentType;
    private final InputStream requestData;

    public MockPortletActionRequest(
            int requestLength, ByteArrayInputStream byteArrayInputStream, String contentType) {
        this.requestData = byteArrayInputStream;
        length = requestLength;
        this.contentType = contentType;
        attributes.put(FileUploadBase.CONTENT_TYPE, contentType);
    }

    public static MockPortletActionRequest MockPortletActionRequest1(
            final byte[] requestData, final String contentType) {
        return new MockPortletActionRequest(
                requestData.length, new ByteArrayInputStream(requestData), contentType);
    }

    public Object getAttribute(String key) {
        return attributes.get(key);
    }

    public Enumeration getAttributeNames() {
        return attributes.keys();
    }

    public String getAuthType() {
        return null;
    }

    public String getContextPath() {
        return null;
    }

    public Locale getLocale() {
        return Locale.getDefault();
    }

    public Enumeration getLocales() {
        return Collections.enumeration(Arrays.asList(Locale.getAvailableLocales()));
    }

    public String getParameter(String key) {
        return parameters.get(key);
    }

    public Map getParameterMap() {
        return Collections.unmodifiableMap(parameters);
    }

    public Enumeration getParameterNames() {
        return Collections.enumeration(parameters.keySet());
    }

    public String[] getParameterValues(String arg0) {
        return null;
    }

    public Enumeration getProperties(String arg0) {
        return null;
    }

    public String getProperty(String arg0) {
        return null;
    }

    public Enumeration getPropertyNames() {
        return null;
    }

    public String getRemoteUser() {
        return null;
    }

    public String getRequestedSessionId() {
        return null;
    }

    public String getResponseContentType() {
        return null;
    }

    public Enumeration getResponseContentTypes() {
        return null;
    }

    public String getScheme() {
        return null;
    }

    public String getServerName() {
        return null;
    }

    public int getServerPort() {
        return 0;
    }

    public boolean isRequestedSessionIdValid() {
        return false;
    }

    public boolean isSecure() {
        return false;
    }

    public boolean isUserInRole(String arg0) {
        return false;
    }

    public void removeAttribute(String key) {
        attributes.remove(key);
    }

    public void setAttribute(String key, Object value) {
        attributes.put(key, value);
    }

    public String getCharacterEncoding() {
        return characterEncoding;
    }

    public int getContentLength() {
        return length;
    }

    public String getContentType() {
        return contentType;
    }

    public InputStream getPortletInputStream() throws IOException {
        return requestData;
    }

    public BufferedReader getReader() throws UnsupportedEncodingException, IOException {
        return null;
    }

    public void setCharacterEncoding(String characterEncoding) throws UnsupportedEncodingException {
        this.characterEncoding = characterEncoding;
    }
}

File: src/isolated_validation/commons-fileupload-python/src/test/org/apache/commons/fileupload/portlet/MockPortletActionRequest.py
from __future__ import annotations
import re
import unittest
import pytest
import pathlib
from io import StringIO
import io
from io import BytesIO
import typing
from typing import *
import locale
from src.main.org.apache.commons.fileupload.FileUploadBase import *


class MockPortletActionRequest:
    
    def setCharacterEncoding(self, characterEncoding: str) -> None:
        self.__characterEncoding = characterEncoding

    def getReader(self) -> io.BufferedReader:
        return io.BufferedReader(io.BytesIO())

    def getPortletInputStream(
        self,
    ) -> typing.Union[io.BytesIO, io.StringIO, io.BufferedReader]:
        return self.__requestData

    def getContentType(self) -> str:
        return self.__contentType

    def getContentLength(self) -> int:
        return self.__length

    def getCharacterEncoding(self) -> str:
        return self.__characterEncoding

    def setAttribute(self, key: str, value: typing.Any) -> None:
        self.__attributes[key] = value

    def removeAttribute(self, key: str) -> None:
        if key in self.__attributes:
            del self.__attributes[key]

    def isUserInRole(self, arg0: str) -> bool:
        return False

    def isSecure(self) -> bool:
        return False

    def isRequestedSessionIdValid(self) -> bool:
        return False

    def getServerPort(self) -> int:
        return 0

    def getServerName(self) -> str:
        return None

    def getScheme(self) -> str:
        return None

    def getResponseContentTypes(
        self,
    ) -> typing.Union[
        typing.Iterator[typing.Any],
        typing.Generator[typing.Any, typing.Any, typing.Any],
    ]:
        return None

    def getResponseContentType(self) -> str:
        return None

    def getRequestedSessionId(self) -> str:
        return None

    def getRemoteUser(self) -> str:
        return None

    def getPropertyNames(
        self,
    ) -> typing.Union[
        typing.Iterator[typing.Any],
        typing.Generator[typing.Any, typing.Any, typing.Any],
    ]:
        return None

    def getProperty(self, arg0: str) -> str:
        return None

    def getProperties(self, arg0: str) -> typing.Union[
        typing.Iterator[typing.Any],
        typing.Generator[typing.Any, typing.Any, typing.Any],
    ]:
        return None

    def getParameterValues(self, arg0: str) -> typing.List[typing.List[str]]:
        return None

    def getParameterNames(
        self,
    ) -> typing.Union[
        typing.Iterator[typing.Any],
        typing.Generator[typing.Any, typing.Any, typing.Any],
    ]:
        return iter(self.__parameters.keys())

    def getParameterMap(self) -> typing.Dict[typing.Any, typing.Any]:
        return self.__parameters

    def getParameter(self, key: str) -> str:
        return self.__parameters.get(key)

    def getLocales(
        self,
    ) -> typing.Union[
        typing.Iterator[typing.Any],
        typing.Generator[typing.Any, typing.Any, typing.Any],
    ]:

        return iter(locale.locale_alias.keys())

    def getLocale(self) -> typing.Any:
        return locale.getdefaultlocale()

    def getContextPath(self) -> str:
        return None

    def getAuthType(self) -> str:
        return None

    def getAttributeNames(
        self,
    ) -> typing.Union[
        typing.Iterator[typing.Any],
        typing.Generator[typing.Any, typing.Any, typing.Any],
    ]:
        return iter(self.__attributes.keys())

    def getAttribute(self, key: str) -> typing.Any:
        return self.__attributes.get(key)

    @staticmethod
    def MockPortletActionRequest1(
        requestData: typing.List[int], contentType: str
    ) -> MockPortletActionRequest:

        byteArrayInputStream = io.BytesIO(bytes(requestData))
        requestLength = len(requestData)

        return MockPortletActionRequest(
            requestLength, byteArrayInputStream, contentType
        )

    def __init__(
        self,
        requestLength: int,
        byteArrayInputStream: typing.Union[io.BytesIO, bytearray],
        contentType: str,
    ) -> None:

        self.__attributes = {}
        self.__parameters = {}
        self.__characterEncoding = None
        self.__length = requestLength
        self.__contentType = contentType
        self.__attributes[FileUploadBase.CONTENT_TYPE] = contentType

        if isinstance(byteArrayInputStream, io.BytesIO):
            self.__requestData = byteArrayInputStream
        else:
            self.__requestData = io.BytesIO(byteArrayInputStream)
